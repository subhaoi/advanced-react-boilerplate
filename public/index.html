<!DOCTYPE html>
<html>
  <head>
    <base href="/" />
    <meta charset="utf-8" />
    <title>React Starter</title>
    <link rel="shortcut icon" href="favicon.ico" />
  </head>

  <body>
    <div id="app"></div>
    <script src="index.js"></script>
  </body>
</html>
