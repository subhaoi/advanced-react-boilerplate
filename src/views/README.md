# Views

All public and private view can be found and managed in the views folder. Views correspond with the routes folder in order to make a view published and viewable on that app when live.