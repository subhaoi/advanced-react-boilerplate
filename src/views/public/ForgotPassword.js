import React, { useState } from 'react';
import { Link } from 'react-router-dom';

export function ForgotPasswordView () {
  const [state, setState] = useState({
    email: '',
  });

  function _handleChange (e) {
    e.persist();
    setState(prevState => ({ ...prevState, [e.target.name]: e.target.value }));
  }

  function _handleSubmit (e) {
    e.preventDefault();
  }

  return (
    <form onSubmit={ _handleSubmit }>
      <fieldset>
        <label htmlFor="view-forgotPassword--email">
          Email
          <input
            id="view-forgotPassword--email"
            type="email"
            name="email"
            value={ state.email }
            onChange={ _handleChange }
            required
          />
        </label>
      </fieldset>
      <button type="submit">Submit</button>
      <ul>
        <li>
          <Link to="/login">Login</Link>
        </li>
      </ul>
    </form>
  );
}
