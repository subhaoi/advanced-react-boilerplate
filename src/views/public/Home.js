import React from 'react';

export function HomeView () {
  return <div>Home</div>;
}
