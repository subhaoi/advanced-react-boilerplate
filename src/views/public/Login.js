import React, { useState } from 'react';
import { Link } from 'react-router-dom';

export function LoginView () {
  const [state, setState] = useState({
    email: '',
    password: '',
  });

  function _handleChange (e) {
    e.persist();
    setState(prevState => ({ ...prevState, [e.target.name]: e.target.value }));
  }

  function _handleSubmit (e) {
    e.preventDefault();
  }

  return (
    <form onSubmit={ _handleSubmit }>
      <fieldset>
        <div>
          <label htmlFor="view-login--email">
            Email
            <input
              id="view-login--email"
              type="email"
              name="email"
              value={ state.email }
              onChange={ _handleChange }
              required
            />
          </label>
        </div>
        <div>
          <label htmlFor="view-login--password">
            Password
            <input
              id="view-login--password"
              type="password"
              name="password"
              value={ state.password }
              onChange={ _handleChange }
              required
            />
          </label>
        </div>
      </fieldset>
      <button type="submit">Login</button>
      <ul>
        <li>
          <Link to="/register">Register</Link>
        </li>
        <li>
          <Link to="/forgot-password">Forgot Password</Link>
        </li>
      </ul>
    </form>
  );
}
