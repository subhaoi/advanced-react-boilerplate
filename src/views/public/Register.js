/* eslint-disable jsx-a11y/label-has-for */
/* eslint-disable jsx-a11y/label-has-associated-control */
import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import { FormInputEmail, FormInputPassword } from '../../components';

export function RegisterView () {
  const [state, setState] = useState({
    email: '',
    password: '',
    confirmPassword: '',
  });

  function _handleChange (e) {
    e.persist();
    setState(prevState => ({ ...prevState, [e.target.name]: e.target.value }));
  }

  function _handleSubmit (e) {
    e.preventDefault();
  }

  return (
    <form onSubmit={ _handleSubmit }>
      <fieldset>
        <div>
          <div>
            <label htmlFor="emailInput">Email</label>
          </div>
          <FormInputEmail
            value={ state.email }
            handleChange={ _handleChange }
          />
        </div>
        <div>
          <div>
            <label htmlFor="passwordInput">Password</label>
          </div>
          <FormInputPassword
            value={ state.password }
            handleChange={ _handleChange }
          />
        </div>
        <div>
          <label htmlFor="view-register--confirmPassword">
            Confirm Password
            <input
              id="view-register--confirmPassword"
              type="password"
              name="confirmPassword"
              value={ state.confirmPassword }
              onChange={ _handleChange }
              required
            />
          </label>
        </div>
      </fieldset>
      <button type="submit">Register</button>
      <ul>
        <li>
          <Link to="/login">Login</Link>
        </li>
        <li>
          <Link to="/forgot-password">Forgot Password</Link>
        </li>
      </ul>
    </form>
  );
}
