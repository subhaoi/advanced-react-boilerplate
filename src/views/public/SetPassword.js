import React, { useState } from 'react';

export function SetPasswordView () {
  const [state, setState] = useState({
    password: '',
  });

  function _handleChange (e) {
    e.persist();
    setState(prevState => ({ ...prevState, [e.target.name]: e.target.value }));
  }

  function _handleSubmit (e) {
    e.preventDefault();
  }

  return (
    <form onSubmit={ _handleSubmit }>
      <fieldset>
        <label htmlFor="setPasswordEmail">
          Email
          <input
            id="setPasswordEmail"
            type="password"
            name="password"
            value={ state.password }
            onChange={ _handleChange }
            required
          />
        </label>
      </fieldset>
      <button type="submit">Submit</button>
    </form>
  );
}
