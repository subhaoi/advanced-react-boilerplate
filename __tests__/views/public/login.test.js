import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { LoginView } from '../../../src/views/public';

it('renders without crashing', () => {
  const component = renderer.create(
    <Router>
      <LoginView />
    </Router>,
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
