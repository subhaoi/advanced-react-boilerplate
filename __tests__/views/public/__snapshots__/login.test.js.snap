// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<form
  onSubmit={[Function]}
>
  <fieldset>
    <div>
      <label
        htmlFor="view-login--email"
      >
        Email
        <input
          id="view-login--email"
          name="email"
          onChange={[Function]}
          required={true}
          type="email"
          value=""
        />
      </label>
    </div>
    <div>
      <label
        htmlFor="view-login--password"
      >
        Password
        <input
          id="view-login--password"
          name="password"
          onChange={[Function]}
          required={true}
          type="password"
          value=""
        />
      </label>
    </div>
  </fieldset>
  <button
    type="submit"
  >
    Login
  </button>
  <ul>
    <li>
      <a
        href="/register"
        onClick={[Function]}
      >
        Register
      </a>
    </li>
    <li>
      <a
        href="/forgot-password"
        onClick={[Function]}
      >
        Forgot Password
      </a>
    </li>
  </ul>
</form>
`;
